### DISCUSSION TITLE
Support for Offline / Air-gapped deployments

### DISCUSSION THREAD

**Discussion by cebcebceb** [Budibase/budibase#3162] [Help]
Hi, we have tried to deploy a Budibase instance with BudiCLI and docker compose on Ubuntu 20.04 LTS for an environment that will not have internet access either on the docker-host with all the Budi-containers or on the webbrowsers that will connect to the Budibase frontend to create apps or use them.

We first deployed the containers on an environment with necessary internet access and exported and imported the VM to the offline environment.

The main problem we have is that on the Webbrowser that lack internet access we get "There was an error loading your app"..."The Budibase client library could not be loaded. Try republishing your app."

While observing these errors we can see timeouts to cdn.jsdeliver.net and fonts.googleapis.com among other things.

As soon as we connect to the Buidbase frontend via a Webbrowser <EDIT: with internet> this issue is resolved.

The problem is that in the environment where we need to have this deployment the NO Internet on the server or the client isn't up for discussion.

So is this scenario at all supported? Is there some way to import said libraries so that they are served by the Budibase server to the clients webbrowser or is internet connection a must on the webbrowser accessing the Budibase apps?

Thank you in advance!

**Comment by mjashanks** [comment]
Hi There. We definitely want to support this scenario. So, thank you for reporting it - I will investigate.

**Reply by cebcebceb** [reply]
Hi, thanks! Happy to hear, we really like what you have done so far, and would be very happy to be able to use Budibase in this environment :)

**Reply by mjashanks** [reply]
Thank you! I logged an issue here: https://github.com/Budibase/budibase/issues/3164

**Comment by tavrez** [comment]
Just a quick question about this, will licensing work on air-gapped installations?
